include/sundials/sundials_errors.h

```diff
@@ -0,0 +1,198 @@
+  ENTRY(SUN_ERR_ARG_CORRUPT, "argument provided is NULL or corrupted")         \
+  ENTRY(SUN_ERR_ARG_INCOMPATIBLE, "argument provided is not comptaible")       \
+  ENTRY(SUN_ERR_ARG_OUTOFRANGE, "argument is out of the valid range")          \
+  ENTRY(SUN_ERR_ARG_WRONGTYPE, "argument provided is not the right type")      \
+  ENTRY(SUN_ERR_ARG_DIMSMISMATCH, "argument dimensions do not agree")          \
+  ENTRY(SUN_ERR_HYPRE_GENERIC, "a generic error ocurred in hypre")             \
+  ENTRY(SUN_ERR_HYPRE_MEMORY, "a memory error ocurred in hypre")               \
+  ENTRY(SUN_ERR_HYPRE_ARG, "an argument error ocurred in hypre")               \
+  ENTRY(SUN_ERR_HYPRE_CONV, "a convergence error ocurred in hypre")            \
+  ENTRY(SUN_ERR_LOGGER_CORRUPT, "SUNLogger is NULL or corrupt")                \
+  ENTRY(SUN_ERR_LOGGER_CANNOTOPENFILE,                                         \
+        "File provided to SUNLogger could not be opened")                      \
+  ENTRY(SUN_ERR_MALLOC_FAIL, "malloc returned NULL")                           \
+  ENTRY(SUN_ERR_MANYVECTOR_COMMNOTSAME,                                        \
+        "not all subvectors have the same MPI_Comm")                           \
+  ENTRY(SUN_ERR_MANYVECTOR_COMMNULL, "MPI_Comm is NULL")                       \
+  ENTRY(SUN_ERR_MPI_FAIL,                                                      \
+        "the MPI call returned something other than MPI_SUCCESS")              \
+  ENTRY(SUN_ERR_NOT_IMPLEMENTED,                                               \
+        "operation is not implemented: function pointer is NULL")              \
+  ENTRY(SUN_ERR_PROFILER_MAPFULL,                                              \
+        "too many SUNProfiler entries, try setting SUNPROFILER_MAX_ENTRIES "   \
+        "in environment to a bigger number")                                   \
+  ENTRY(SUN_ERR_PROFILER_MAPGET, "unknown error getting SUNProfiler timer")    \
+  ENTRY(SUN_ERR_PROFILER_MAPINSERT,                                            \
+        "unknown error inserting SUNProfiler timer")                           \
+  ENTRY(SUN_ERR_PROFILER_MAPKEYNOTFOUND, "timer was not found in SUNProfiler") \
+  ENTRY(SUN_ERR_PROFILER_MAPSORT, "error sorting SUNProfiler map")             \
+  ENTRY(SUN_ERR_SUNCTX_CORRUPT, "SUNContext is NULL or corrupt")               \
+  ENTRY(SUN_ERR_GENERIC, "")                                                   \
+  ENTRY(SUN_ERR_UNKNOWN, "Unknown error occured: open an issue at "            \
+                         "https://github.com/LLNL/sundials")
+
+/* Expand SUN_ERR_CODE_LIST to enum */
+#define SUN_EXPAND_TO_ENUM(name, description) name,
+
+/* SUNErrorCode range is [-1000, -2000] to avoid conflicts wiht package error
+   codes, and old/deprecated codes for matrix and (non)linear solvers. */
+enum
+{
+  SUN_ERR_MINIMUM                                       = -2000,
+  SUN_ERR_CODE_LIST(SUN_EXPAND_TO_ENUM) SUN_ERR_MAXIMUM = -1000,
```

Comment:
Just above here, it looks like `SUN_EXPAND_TO_ENUM` is a macro that requires 2 arguments, but here it is used with none?